src/hotspot/share/ci/ciReplay.cpp

```diff
@@ -115,6 +115,7 @@ class CompileReplay : public StackObj {
   Handle  _protection_domain;
   bool    _protection_domain_initialized;
   Handle  _loader;
+  int     _version;
```

Comment:
You forgot to initialize `_version` to 0. Otherwise, it could contain garbage for old replay files without version number (possibly `> REPLAY_VERSION`).